server/venueless/api/views.py

```diff
@@ -115,6 +119,90 @@ def get(self, request, **kwargs):
             )
 
 
+class CreateWorldView(APIView):
+    authentication_classes = []  # disables authentication
+    permission_classes = []
+
+    @staticmethod
+    def post(request, *args, **kwargs) -> JsonResponse:
+        payload = CreateWorldView.get_payload_from_token(request)
+
+        # check if user has permission to create world
+        if payload.get("has_permission"):
+            secret = get_random_string(length=64)
+            config = {
+                "JWT_secrets": [
+                    {
+                        "issuer": "any",
+                        "audience": "venueless",
+                        "secret": secret,
+                    }
+                ]
+            }
+
+            title_default = {
+                value for key, value in request.data.get("title").items() if value
+            }.pop()
+
+            # if world already exists, update it, otherwise create a new world
+            try:
+                if World.objects.filter(id=request.data.get("id")).exists():
```

Comment:
Data received from outside work (form, HTTP request) have to be validated.

You are calling 
```py
World.objects.get(id=request.data.get("id")
```

but what if some attackers pass "id=abc"? This code will throw exception.